### PR TITLE
[tempershow] Add SFP transceiver temperature support with DOM table fallback

### PR DESCRIPTION
**Pull request by vvolam**
#### What I did

Added SFP/transceiver temperature display support to the `tempershow` utility (used by `show platform temperature` command). The change enables users to view transceiver module temperatures alongside existing platform sensors.

#### How I did it

1. Enhanced `tempershow` script to read transceiver temperature data from STATE_DB
2. Implemented fallback logic:
   - First tries `TRANSCEIVER_DOM_TEMPERATURE` table (preferred/new table)
   - Falls back to `TRANSCEIVER_DOM_SENSOR` table if temperature not found in the first table
3. Added helper method `_get_transceiver_temperature_data()` for clean abstraction of the fallback logic
4. Transceiver temperatures are displayed with sensor names based on port names (e.g., Ethernet0, Ethernet4)
5. Added comprehensive unit tests covering:
   - Platform-only temperature data
   - Transceiver-only temperature data  
   - Combined platform and transceiver data
   - Fallback from TRANSCEIVER_DOM_TEMPERATURE to TRANSCEIVER_DOM_SENSOR
   - JSON output format
   - Edge cases (empty tables, missing fields)

#### How to verify it

1. Run unit tests:
   ```bash
   python3 -m pytest tests/tempershow_test.py -v
   ```
2. On a SONiC device with SFP modules installed:
# Verify tables have data
sonic-db-cli STATE_DB keys "TEMPERATURE_INFO|*"
sonic-db-cli STATE_DB keys "TRANSCEIVER_DOM_TEMPERATURE|*"
sonic-db-cli STATE_DB keys "TRANSCEIVER_DOM_SENSOR|*"

# Test the command
show platform temperature
show platform temperature -j
#### Previous command output (if the output of a command-line utility has changed)
image

#### New command output (if the output of a command-line utility has changed)
image
.
.

### CODE REVIEW

File: scripts/tempershow [hunk 1]
```diff
@@ -28,53 +30,106 @@ class TemperShow(object):
...
+        transceiver_keys = self.db.keys(self.db.STATE_DB, TRANSCEIVER_DOM_TEMPERATURE_TABLE_NAME + '*')
+        if not transceiver_keys:
+            transceiver_keys = self.db.keys(self.db.STATE_DB, TRANSCEIVER_DOM_SENSOR_TABLE_NAME + '*')
+
+        for key in natsorted(transceiver_keys or []):
+            key_list = key.split('|')
+            if len(key_list) != 2:
+                continue
+
+            port_name = key_list[1]
```
[the remark this answers is not in the document]
Comment (vvolam) [reply scripts/tempershow]: not needed. won't fix

File: scripts/tempershow [hunk 2]
```diff
@@ -28,53 +30,106 @@ class TemperShow(object):
...
+            temp_value = self._get_transceiver_temperature_data(port_name)
+            if temp_value:
+                self._add_sensor_to_output(port_name, temp_value, None, json_output, table)
```
Comment (Junchao-Mellanox) [inline scripts/tempershow]: Two questions here:
1. how about threshold data?
2. The format of the command was:
```
admin@sonic:~$ show plat temp
                Sensor    Temperature    High TH    Low TH    Crit High TH    Crit Low TH    Warning          Timestamp
----------------------  -------------  ---------  --------  --------------  -------------  ---------  -----------------
...
    xSFP module 1 Temp            N/A        N/A       N/A             N/A            N/A      False  20260129 05:55:35
    xSFP module 2 Temp           45.0       70.0       N/A            75.0            N/A      False  20260129 05:55:35
```

But after this change, I believe it will become:
```
admin@sonic:~$ show plat temp
                Sensor    Temperature    High TH    Low TH    Crit High TH    Crit Low TH    Warning          Timestamp
----------------------  -------------  ---------  --------  --------------  -------------  ---------  -----------------
...
    Ethernet0            N/A        N/A       N/A             N/A            N/A      False  20260129 05:55:35
    Ethernet8            45.0       70.0       N/A            75.0            N/A      False  20260129 05:55:35
```
It will break backward compatible.

Comment (vvolam) [reply scripts/tempershow]: I could read threshold data as well. will update the patch.

Format of the command will not change. Here is the direct screenshot instead of the copy

image

Comment (Junchao-Mellanox) [reply scripts/tempershow]: Could you please check the old format for Nvidia platform? I think the format will change after your PR.

Comment (vvolam) [reply scripts/tempershow]: I have verified the format on 5640 platform and don't observe the formatting issue

image

Comment (vvolam) [reply scripts/tempershow]: Added threshold data and revised format changes. Should be good now!

Comment (Junchao-Mellanox) [reply scripts/tempershow]: Here is original output: 
c07e72ad-74bb-4ab3-bb71-959964362d82

Comment (vvolam) [reply scripts/tempershow]: Ok you were pointing to the Sensor values, As you see, we are updating these names read as-is from the tables.

admin@XXX-sn5640-X:~$ sonic-db-cli STATE_DB keys "TEMPERATURE_INFO|*" | grep SFP | head -n 5
TEMPERATURE_INFO|xSFP module 41 Temp
TEMPERATURE_INFO|xSFP module 30 Temp
TEMPERATURE_INFO|xSFP module 63 Temp
TEMPERATURE_INFO|xSFP module 64 Temp
TEMPERATURE_INFO|xSFP module 51 Temp
admin@XXX-sn5640-X:~$ sonic-db-cli STATE_DB keys "TRANSCEIVER_DOM_SENSOR|*" | head -n 5
TRANSCEIVER_DOM_SENSOR|Ethernet352
TRANSCEIVER_DOM_SENSOR|Ethernet336
TRANSCEIVER_DOM_SENSOR|Ethernet360
TRANSCEIVER_DOM_SENSOR|Ethernet344
TRANSCEIVER_DOM_SENSOR|Ethernet40

If the entries in TRANSCEIVER_DOM_TEMPERATURE is same as TEMPERATURE_INFO, command also displays same..

File: scripts/tempershow [hunk 3]
```diff
@@ -28,53 +31,99 @@ class TemperShow(object):
...
+        if not transceiver_keys:
+            transceiver_keys = self.db.keys(self.db.STATE_DB, TRANSCEIVER_DOM_SENSOR_TABLE_NAME + '*')
+
+        for key in natsorted(transceiver_keys or []):
+            key_list = key.split('|')
+            if len(key_list) != 2:
+                continue
+
+            port_name = key_list[1]
+            temp_data = self._get_transceiver_temperature_data(port_name)
```
Comment (judyjoseph) [inline scripts/tempershow]: An optimization here -- since we know the key here, can we frame the key string here and pass to this function _get_transceiver_temperature_data()

Comment (vvolam) [reply scripts/tempershow]: Optimized it further. Thank you

**Comment by vvolam** [comment]
> Hi @vvolam, @judyjoseph , as I mentioned before, this change breaks backward compatible:
> 
> c44e5477-09b9-4a82-8490-4aec25ff9761
> Previously, the name for module temp sesor is the "xSFP module 1 Temp", but now it becomes "Ethernet0".

Hi @Junchao-Mellanox,

We believe using “EthernetXX” aligns better with SONiC interface naming and makes mapping more intuitive.

Could you please let us know if any of your tools currently depend on the “xSFP module 1 Temp” naming?

File: scripts/tempershow [hunk 4]
```diff
@@ -28,53 +31,75 @@ class TemperShow(object):
...
+        # Collect thermal sensor data from TEMPERATURE_INFO table
+        thermal_keys = self.db.keys(self.db.STATE_DB, TEMPER_TABLE_NAME + '*')
+        self._process_keys(natsorted(thermal_keys or []), json_output, table)
+
+        # Collect transceiver temperature data for SFP modules
+        transceiver_keys = self.db.keys(self.db.STATE_DB, TRANSCEIVER_DOM_TEMPERATURE_TABLE_NAME + '*')
+        if not transceiver_keys:
+            transceiver_keys = self.db.keys(self.db.STATE_DB, TRANSCEIVER_DOM_SENSOR_TABLE_NAME + '*')
+
+        self._process_keys(natsorted(transceiver_keys or []), json_output, table)
```
Comment (judyjoseph) [inline scripts/tempershow:90]: this can be inside if not block ?

**Comment by Junchao-Mellanox** [comment]
Hi @judyjoseph , I sent an email to describe the potential impaction. Could you please share your comments to that email? And changes that break backward compatible will need production team and management team approval.

File: scripts/tempershow (hunk 4 above)
Comment (vvolam) [reply scripts/tempershow:90]: no, in either case, we receive transceiver_keys to process for display.

**Comment by vvolam** [comment]
Instead of modifying the CLI, we could read the values in thermalctld from STATE_DB to have backward compatibility, Changes are done with https://github.com/sonic-net/sonic-platform-daemons/pull/747
